lib/deploy/stepFunctions/compileIamRole.js

```diff
@@ -70,22 +70,58 @@ function sqsQueueUrlToArn(serverless, queueUrl) {
   return [];
 }
 
-function getSqsPermissions(serverless, state) {
+function getSqsPermissions(serverless, state, action = 'sqs:SendMessage') {
   if (_.has(state, 'Parameters.QueueUrl')
       || _.has(state, ['Parameters', 'QueueUrl.$'])) {
     // if queue URL is provided by input, then need pervasive permissions (i.e. '*')
     const queueArn = state.Parameters['QueueUrl.$']
       ? '*'
       : sqsQueueUrlToArn(serverless, state.Parameters.QueueUrl);
-    return [{ action: 'sqs:SendMessage', resource: queueArn }];
+    return [{ action, resource: queueArn }];
   }
   logger.log('SQS task missing Parameters.QueueUrl or Parameters.QueueUrl.$');
   return [];
 }
 
+function getSsmPermissions() {
+  return {
+    action: 'ssm:GetParameters*',
+    resource: [
+      {
+        'Fn::Join': [
+          ':',
+          [
+            'arn:aws:ssm',
+            { Ref: 'AWS::Region' },
+            { Ref: 'AWS::AccountId' },
+            'parameter/*',
+          ],
+        ],
+      }],
+  };
+}
+
+function getApiGatewayPermissions() {
+  return [{
+    action: 'execute-api:Invoke',
+    resource: [
+      {
+        'Fn::Join': [
+          ':',
+          [
+            'arn:aws:execute-api',
+            { Ref: 'AWS::Region' },
+            { Ref: 'AWS::AccountId' },
+            '*',
```

Comment:
I don't know how to get the actual api ID. I'm not sure if we need permissions that restrictive.